api/dataprocessor_test.go

```diff
@@ -1041,14 +1041,54 @@ func TestAlignRequests(t *testing.T) {
 			t.Errorf("different amount of requests for testcase %d  expected: %v, got: %v", i, len(ac.outReqs), len(out))
 		} else {
 			for r, exp := range ac.outReqs {
-				if exp != out[r] {
+				if !compareReqEqual(exp, out[r]) {
 					t.Errorf("testcase %d, request %d:\nexpected: %v\n     got: %v", i, r, exp.DebugString(), out[r].DebugString())
 				}
 			}
 		}
 	}
 }
 
+func compareReqEqual(a, b models.Req) bool {
```

Comment:
I am hesitant to add a Equal method to Req as it is only needed for this one test.
The goal is to just compare to compare alignCase.reqs with alignCase.outReqs.  This was done with a `==`, but now that Req.Node is no longer a pointer you cant do that.   If these tests used github.com/smartystreets/goconvey/convey  we would just use ShouldResemble to do a deep comparision.